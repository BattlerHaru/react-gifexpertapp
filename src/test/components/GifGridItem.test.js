const { shallow } = require("enzyme");
import { GifGridItem } from "../../components/GifGridItem";

describe("Pruebas en <GifGridItem>", () => {
  const title = "title example";
  const url = "https://example.com/images/example.png";

  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("Debe de mostrar el componente correctametne", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de tener un parrafo con el title", () => {
    const p = wrapper.find("p").text().trim();
    expect(p).toBe(title);
  });

  test("Debe de tener una img con el url y el alt de los pops", () => {
    const img = wrapper.find("img");
    // console.log(img.html());
    // console.log(img.props());
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("Debe de tener animate__fadeIn", () => {
    const div = wrapper.find("div");
    const className = div.prop("className");
    expect(className.includes("animate__fadeIn")).toBeTruthy();
  });
});
