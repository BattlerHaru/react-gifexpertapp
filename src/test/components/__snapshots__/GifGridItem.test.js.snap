// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem> Debe de mostrar el componente correctametne 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="title example"
    src="https://example.com/images/example.png"
  />
  <p>
    title example
  </p>
</div>
`;
